src/modules/imselector/imselector.cpp

```diff
@@ -12,6 +12,27 @@
 
 namespace fcitx {
 
+bool selectInputMethod(InputContext *ic, IMSelector *imSelector,
```

Comment:
Actually I mean, only fcitx::Instance is required here. no need to wrap it with IMSelector

Reply:
How can we access `auto *state = ic->propertyFor(&imSelector->factory())` then?